haskell-src/Concordium/Genesis/Data/P8.hs

```diff
@@ -14,6 +14,34 @@ import qualified Concordium.Crypto.SHA256 as Hash
 import qualified Concordium.Genesis.Data.Base as Base
 import qualified Concordium.Genesis.Data.BaseV1 as BaseV1
 import Concordium.Types
+import Concordium.Types.Parameters
+
+-- | Parameters data type for the 'P7' to 'P8' protocol update.
+--  This is provided as a parameter to the protocol update chain update instruction.
+newtype ProtocolUpdateData = ProtocolUpdateData
+    { -- | The 'ValidatorScoreParameters' that the protocol should
+      --  be instantiated with.
+      updateValidatorScoreParameters :: ValidatorScoreParameters
+    }
+    deriving (Eq, Show)
+
+instance Serialize ProtocolUpdateData where
+    put ProtocolUpdateData{..} = do
+        put updateValidatorScoreParameters
+    get = do
+        updateValidatorScoreParameters <- get
+        return ProtocolUpdateData{..}
+
+-- | Parameters used to migrate state from 'P7' to 'P8'.
+data StateMigrationData = StateMigrationData
+    { -- | Data provided by the protocol update to be used
+      --  in the migration.
+      migrationProtocolUpdateData :: !ProtocolUpdateData,
+      -- | The timestamp for the first trigger block in the new consensus.
+      --  This should be set to one epoch past the regenesis time.
+      migrationTriggerBlockTime :: !Timestamp
```

Comment:
Is this needed anywhere?

Reply:
I assumed this is a field needed in the migration mechanism. But maybe not. I removed it, the build works. Let's see if the tests work as well.